### PR TITLE
[MM-1025]: Fixed the infinite loading state of tooltip modal

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**
### Summary
Fixed the infinite loading state of the tooltip modal
#### Existing
Screenshot from 2024-09-20 16-10-59
#### Updated
Screenshot from 2024-09-23 15-15-59

### CODE REVIEW

File: webapp/src/components/jira_ticket_tooltip/jira_ticket_tooltip.tsx [hunk 1]
```diff
@@ -166,7 +173,21 @@ export default class TicketPopover extends React.PureComponent<Props, State> {
             return null;
         }
 
-        const {ticketDetails} = this.state;
+        const {ticketDetails, error} = this.state;
+        if (error) {
+            return (
+                <div className='jira-issue-tooltip jira-issue-tooltip-error'>
+                    <span
+                        className='jira-issue-error-icon fa fa-exclamation-triangle'
```
Comment (raghavaggarwal2308) [inline webapp/src/components/jira_ticket_tooltip/jira_ticket_tooltip.tsx]: Use the icon showed the designs

File: webapp/src/components/jira_ticket_tooltip/ticketStyle.scss [hunk 2]
```diff
@@ -231,3 +231,22 @@
         opacity: 0.72;
     }
 }
+
+.jira-issue-tooltip-error {
```
Comment (raghavaggarwal2308) [inline webapp/src/components/jira_ticket_tooltip/ticketStyle.scss:235]: Add some padding

Comment (raghavaggarwal2308) [reply webapp/src/components/jira_ticket_tooltip/ticketStyle.scss:235]: Also get this CSS reviewed from @avas27JTG

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

File: webapp/src/components/jira_ticket_tooltip/jira_ticket_tooltip.tsx (hunk 1 above)
Comment (Kshitij-Katiyar) [reply webapp/src/components/jira_ticket_tooltip/jira_ticket_tooltip.tsx]: @raghavaggarwal2308 The icon used in the designs is part of the pro version of Font Awesome. I couldn't find a way to use it in our code. Please let me know of any workaround for this.

Comment (raghavaggarwal2308) [reply webapp/src/components/jira_ticket_tooltip/jira_ticket_tooltip.tsx]: You can use an svg for this

File: webapp/src/components/jira_ticket_tooltip/ticketStyle.scss (hunk 2 above)
Comment (Kshitij-Katiyar) [reply webapp/src/components/jira_ticket_tooltip/ticketStyle.scss:235]: Hey @avas27JTG, can you please review these css

File: webapp/src/components/plugin_constants/icons.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+export const SVGIcons: Record<SvgIconNames, JSX.Element> = {
+    exclamationTriangle: (
+        <svg
+            xmlns='http://www.w3.org/2000/svg'
+            width='30'
+            height='30'
+            fill='red'
+            className='bi bi-exclamation-triangle'
+            viewBox='0 0 16 16'
+        >
```
Comment (raghavaggarwal2308) [inline webapp/src/components/plugin_constants/icons.tsx]: This is part of the svg wrapper, so it should not be here

File: webapp/src/components/jira_ticket_tooltip/jira_ticket_tooltip.tsx [hunk 4]
```diff
@@ -177,11 +179,14 @@ export default class TicketPopover extends React.PureComponent<Props, State> {
...
                 <div className='jira-issue-tooltip jira-issue-tooltip-error'>
-                    <span
-                        className='jira-issue-error-icon fa fa-exclamation-triangle'
-                        style={{color: 'red'}}
-                        title={'Hazard Icon'}
-                    />
+                    <SVGWrapper
+                        width={30}
+                        height={30}
+                        fill='red'
```
Comment (raghavaggarwal2308) [inline webapp/src/components/jira_ticket_tooltip/jira_ticket_tooltip.tsx]: Use hex code for color

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

File: webapp/src/components/plugin_constants/icons.tsx [hunk 5]
```diff
@@ -4,14 +4,7 @@ type SvgIconNames = 'exclamationTriangle';
...
     exclamationTriangle: (
-        <svg
-            xmlns='http://www.w3.org/2000/svg'
-            width='30'
-            height='30'
-            fill='red'
-            className='bi bi-exclamation-triangle'
-            viewBox='0 0 16 16'
-        >
+        <svg>
```
Comment (raghavaggarwal2308) [inline webapp/src/components/plugin_constants/icons.tsx]: No need of svg as well

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

File: webapp/src/components/plugin_constants/icons.tsx [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+
+type SvgIconNames = 'exclamationTriangle';
+
+export const SVGIcons: Record<SvgIconNames, JSX.Element> = {
+    exclamationTriangle: (
+        <>
```
Comment (raghavaggarwal2308) [inline webapp/src/components/plugin_constants/icons.tsx:7]: What is the meed for this fragment

Comment (raghavaggarwal2308) [inline webapp/src/components/plugin_constants/icons.tsx:7]: What is the need for this empty fragment?

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

File: webapp/src/components/plugin_constants/icons.tsx (hunk 6 above)
Comment (Kshitij-Katiyar) [reply webapp/src/components/plugin_constants/icons.tsx:7]: Function can only return one JSX element, hence the two paths need to be wrapped in some fragment

**Review by raghavaggarwal2308 — APPROVED** [review]

**Comment by Kshitij-Katiyar** [comment]
Created a PR on mattermost remote with all these changes hence closing this one

**Pull request closed without merging**